release/models/bgp/openconfig-bgp-bmp.yang

```diff
@@ -0,0 +1,650 @@
+
+  yang-version "1";
+
+  namespace "http://openconfig.net/yang/bgp-bmp";
+
+  prefix "oc-bgp-bmp";
+
+  import openconfig-bgp { prefix oc-bgp; }
+  import openconfig-bgp-types { prefix oc-bgp-types; }
+  import openconfig-network-instance { prefix oc-netinst; }
+  import openconfig-inet-types { prefix oc-inet; }
+  import openconfig-extensions { prefix oc-ext; }
+  import openconfig-yang-types { prefix oc-yang; }
+
+  organization
+    "OpenConfig working group";
+
+  contact
+    "OpenConfig working group
+    www.openconfig.net";
+
+  description
+    "This module defines configuration and operational state data for
+    BGP Monitoring Protocol (BMP) as defined in RFC 7854.";
+
+  oc-ext:openconfig-version "9.9.0";
+
+  revision "2025-04-21" {
+    description
+      "Initial revision";
+    reference
+      "RFC 7854: BGP Monitoring Protocol";
+  }
+
+  // Typedefs and identities
+  typedef connection-mode-type {
+    type enumeration {
+      enum ACTIVE {
+        description
+          "BMP client actively initiates connections to BMP servers";
+      }
+      enum PASSIVE {
+        description
+          "BMP client passively listens for connections from BMP servers";
+      }
+    }
+    description
+      "Type to define the connection mode for BMP sessions";
+  }
+
+  typedef route-monitoring-policy-type {
+    type enumeration {
+      enum PRE_POLICY {
+        description
+          "Send pre-policy route monitoring messages";
+      }
+      enum POST_POLICY {
+        description
+          "Send post-policy route monitoring messages";
+      }
+      enum BOTH {
+        description
+          "Send both pre-policy and post-policy route monitoring messages";
+      }
+    }
+    description
+      "Type to define the route monitoring policy for BMP";
+  }
+
+  typedef connection-status-type {
+    type enumeration {
+      enum CONNECTED {
+        description
+          "BMP session is established and operational";
+      }
+      enum CONNECTING {
+        description
+          "BMP session is in the process of establishing connection";
+      }
+      enum DISCONNECTED {
+        description
+          "BMP session is not connected";
+      }
+    }
+    description
+      "Type to define the operational status of a BMP session";
+  }
+
+  // Groupings
+
+  grouping bmp-tcp-keepalive-config {
+    description
+      "Configuration parameters for BMP TCP keepalive settings";
+
+    leaf idle-time {
+      type uint16;
+      description
+        "The time in seconds before sending a TCP keepalive probe
+        when the connection is idle";
+    }
+
+    leaf probe-count {
+      type uint8;
+      description
+        "The maximum number of TCP keepalive probes to send before
+        declaring the connection dead";
+    }
+
+    leaf probe-interval {
+      type uint16;
+      description
+        "The time in seconds between TCP keepalive probes";
+    }
+  }
+
+  grouping bmp-tcp-keepalive-state {
```

Comment:
The same grouping can be reused for config/state (reflection) vs. redefining each node again